src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -336,6 +336,13 @@ class BinaryCompactObject : public DomainCreator<3> {
     static constexpr Options::String help = {"Radius of the entire domain."};
   };
 
+  struct CoverCompactObjectsWithCubes {
```

Comment:
(Not a review)

I don't love this name because we technically cover the black holes with cubes as well. It's just with the BHs there's also some inner structure as well (and the cubes are DG not FD). However, I'm struggling to think of a better name:

- `CubesHaveStructure`
- `UseCartesianCubesAroundObjects`
- `CoverObjectsWithStrucutre`

Actually, I think the second name I suggested, `UseCartesianCubesAroundObjects`, is pretty clear. If it's true, it does what it says. If it's false, then the cubes use DG and have some inner structure. What do you think?

Reply:
I'm alright with the alternative name. One thing to note though is that, as far as I understand the code, we won't require the use of FD in the cubes. We just need cubes to allow it for now, but can still use DG in smooth region.

Reply:
Ideally we would replace the options that currently define objects A and B with a simple `CartesianCubeAtXCoord: 10.` or something like that. I think that can be done by changing the type of the `ObjectA` and `ObjectB` options from `Object` to `std::variant<Object, CartesianObject>` and implementing a new `CartesianObject` struct that parses however you want (I think it just needs a single x-coord option). Then you don't have to specify unused options like inner and outer radii for the objects. The result is this:

```yaml
DomainCreator:
  BinaryCompactObject:
    # BHNS domain
    ObjectA:
      CartesianCubeAtXCoord: 20.
    ObjectB:
      InnerRadius: 4.
      OuterRadius: 9.
      XCoord: -20.
      Interior:
        ExciseWithBoundaryCondition: # ...
      UseLogarithmicMap: False
  # ...
```

Reply:
This is true. Maybe `UseSingleCubeAroundObjects`? That way we don't say anything about cartesian because I believe that will be taken care of in the `SubcellOptions`. There you specify which blocks are allowed to do FD and which are restricted to just DG.

Reply:
@knelli2 what's your objection to "Cartesian" in the name? I think it makes clear that the grid is not curved, which is the point of the exercise. Other name suggestions are welcome of course. I don't see the connection to SubcellOptions.

Reply:
Not sure why, but I always associate "cartesian" with "FD" (maybe because for vacuum we always use curved meshes with spectral methods). However, that's obviously wrong because we can do DG on a cartesian grid. So I don't have an objection, just my brain getting things mixed up 🧠 .